src/DynamoCore/Models/NodeLoaders/ZeroTouchNodeLoader.cs

```diff
@@ -112,6 +112,21 @@ public NodeModel CreateNodeFromXml(XmlElement nodeElement, SaveContext context)
                 result = new DSFunction(descriptor);
 
             result.Deserialize(nodeElement, context);
+
+            // In case of input parameters mismatch, use default arguments for parameters that have one
+            if (descriptor.MangledName != function)
+            {
+                string[] inputTypes = function.Split('@')[1].Split(',');
+                var i = 0;
+                foreach (var param in descriptor.Parameters.Select((p, j) => new { p.Type, j }))
+                {
+                    if (i >= inputTypes.Length || param.Type.ToString() != inputTypes[i])
+                        result.InPorts[param.j].UsingDefaultValue = result.InPortData[param.j].DefaultValue != null;
```

Comment:
Here we compare the type, and if type mismatch, then may choose using default argument. What happen if a function is overloaded? For example, 
```
Foo(Point p1 = default point 2, Point p2 = default point 1);
Foo(Line l1, Line l2);
```
And we open a .dyn whose signature is `Foo@Line,Line`, will it be changed to `Foo@Point,Point` and use default argument?